### DISCUSSION TITLE
Downloading files with `models:/<name>@alias/path/to/file`?

### DISCUSSION THREAD

**Discussion by kimminw00** [mlflow/mlflow#14170] [Q&A] [answered]
I was wondering if it's possible to download arbitrary files from the MLflow tracking server using 
 <code>mlflow.download_artifacts("models:/<name>@alias/path/to/file</code>")</code>. 

### Reference
https://github.com/mlflow/mlflow/blob/facda38aebcb458119ac2d9b9636f3b35b4105d2/mlflow/tracking/artifact_utils.py#L88-L96

https://github.com/mlflow/mlflow/blob/facda38aebcb458119ac2d9b9636f3b35b4105d2/mlflow/store/artifact/models_artifact_repo.py#L78-L94

**Comment by SahilKumar75** [comment, marked as answer]
Yes, this works. I went through the source to confirm it.

The `split_models_uri` method in [`models_artifact_repo.py`](https://github.com/mlflow/mlflow/blob/master/mlflow/store/artifact/models_artifact_repo.py) explicitly handles the alias + subpath case. The docstring even calls it out:

Split 'models:/<name>@alias/path/to/model' into
('models:/<name>@alias', 'path/to/model').

The relevant logic is here:

```python
splits = path.split("/", 3)
cut_index = 2 if "@" in splits[1] else 3
model_name_and_version = splits[:cut_index]
artifact_path = "/".join(splits[cut_index:])
return base_part + "/".join(model_name_and_version), artifact_path
```

When it sees `@` in the model name segment it sets `cut_index = 2`, which correctly separates the alias from the rest of the path. Then in `artifact_utils.py` (the code you referenced), it calls `split_models_uri` and uses the resulting `artifact_path` to download only that specific file rather than the whole artifact directory.

So in practice:

```python
import mlflow

# downloads only config/params.json from the version pointed to by 'champion'
local_path = mlflow.artifacts.download_artifacts(
    artifact_uri="models:/my-model@champion/config/params.json"
)

# downloads a nested file
local_path = mlflow.artifacts.download_artifacts(
    artifact_uri="models:/my-model@champion/data/preprocessor.pkl"
)

# downloads the entire model artifact (no subpath)
local_path = mlflow.artifacts.download_artifacts(
    artifact_uri="models:/my-model@champion"
)
```

One thing to be aware of: the path splitting only triggers when there are at least 2 forward slashes after the scheme, i.e. `path.count("/") >= 2`. That condition is satisfied whenever you have at least one path segment after the alias, so any file or folder works fine. If you just pass the bare alias URI with no path it falls through and downloads the whole artifact root, which is the expected behaviour.

This has been in the codebase since alias support was introduced, so it should work on any version that supports `models:/<name>@alias` URIs.
